tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -114,23 +100,11 @@ func indexOf(element string, data []string) int {
 	return -1 // not found.
 }
 
-func (rb *IptablesRuleBuilder) appendInternal(ipt *[]Rule, command iptableslog.Command, chain string, table string, params ...string) *IptablesRuleBuilder {
+func (rb *IptablesRuleBuilder) appendInternal(ipt *[]Rule, chain string, table string, params ...string) *IptablesRuleBuilder {
 	idx := indexOf("-j", params)
 	if idx < 0 && !strings.HasPrefix(chain, "ISTIO_") {
 		log.Warnf("Appending non-jump rule in non-Istio chain (rule: %s) \n", strings.Join(params, " "))
 	}
-	// We have identified the type of command this is and logging is enabled. Appending a rule to log this chain will be hit
-	if rb.cfg.TraceLogging && idx >= 0 && command != iptableslog.UndefinedCommand {
-		match := params[:idx]
-		// 1337 group is just a random constant to be matched on the log reader side
-		// Size of 20 allows reading the IPv4 IP header.
-		match = append(match, "-j", "NFLOG", "--nflog-prefix", command.Identifier, "--nflog-group", "1337", "--nflog-size", "20")
```

Comment:
> but I assumed it would only log packets that jump to that chain. If we don't have jumps to that chain anymore, what will be the default logging behavior?

No jumps, no log.

I've never used this nflog feature either, and I doubt John has. I don't know of any users that rely on it. My contention is that if none of the three of us find it useful for debugging and have never used it, it's not worth the cruft :P

If we have a real need for this, it would be much better to hook into the rule creation logic and insert log rules silently when a jump rule is created, this impl requires everyone adding a rule to know what kind of logging primitive to use for it - and most rules just get UndefinedCommand, which is never logged.

Also, logging only jumps isn't generally useful IMO. I could see an argument for logging exclusions, but we don't do that consistently, and how we do it here isn't particularly useful.

Reply:
I was also the one to originally add it btw. Seemed useful at the time but isn't holding up much value. If we are killing it there is more we can / should cleanup though later

Reply:
> I was also the one to originally add it btw. Seemed useful at the time but isn't holding up much value. 


Ah, fair enough. Thank you for saying so, and I would have probably agreed with you at the time that it seemed useful as well.

> If we are killing it there is more we can / should cleanup though later

100%. I was mostly canvassing to see if anyone who had a use for it that I had missed might speak up, but if not I'm happy to clean up the rest in a followup.

Reply:
Thank you for the clarification.
I'm fine with its removal (+follow-up to cleanup the rest), as I personally would prefer to log into the pod and create logging rules myself if there is the need.
Also, some of those rules end up in the main chains, which are a pain to reconcile.